aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -81,16 +106,55 @@ private GetPolicyResponse getExistingPolicy(AmazonWebServicesClientProxy proxy,
         );
     }
 
+    private void updateNotificationChannels(String pgName, AmazonWebServicesClientProxy proxy, List<Channel> requestedConfiguration) {
+        List<Channel> currentChannels = getExistingNotificationConfiguration(proxy, pgName).notificationConfiguration().channels();
+        updateNotificationChannels(currentChannels, requestedConfiguration, pgName, proxy);
+    }
+
+    private GetNotificationConfigurationResponse getExistingNotificationConfiguration(AmazonWebServicesClientProxy proxy, String profilingGroupName) {
+        return proxy.injectCredentialsAndInvokeV2(GetNotificationConfigurationRequest.builder()
+                        .profilingGroupName(profilingGroupName)
+                        .build(),
+                profilerClient::getNotificationConfiguration);
+    }
+
+    // Generate the change-set and add / delete notification channel based on change-set
+    private void updateNotificationChannels(List<Channel> currentChannels, List<Channel> requestedConfiguration, String pgName, AmazonWebServicesClientProxy proxy) {
+        Map<String, Channel> currentChannelsMap = currentChannels.stream().collect(Collectors.toMap(Channel::uri, val -> val));
+        Map<String, Channel> requestedConfigurationMap = requestedConfiguration.stream().collect(Collectors.toMap(Channel::uri, val -> val));
+
+        for (Channel currentChannel : currentChannels) {
+            if (!requestedConfigurationMap.containsKey(currentChannel.uri())) {
+                // Can assert that channel.id() exists here, since this is from an existingChannel, which is fetched via the GetNotificationConfiguration, which has the id attached
+                NotificationChannelHelper.deleteNotificationChannel(pgName, currentChannel.id(), proxy, profilerClient);
+            } else {
+                // check if there is an updated id, and if so then re-add the channel with the new id, else use the existing channel
+                Channel requestedChannel = requestedConfigurationMap.get(currentChannel.uri());
+                if (requestedChannel.id() != null && !currentChannel.id().equals(requestedChannel.id())) {
+                    NotificationChannelHelper.updateChannel(pgName, currentChannel.id(), requestedChannel, proxy, profilerClient);
```

Comment:
Minor: Perhaps this could be `updateChannelId(pgName, currentChannel, requestedChannel, ...)`?